node/cli/src/service.rs

```diff
@@ -120,7 +109,7 @@ construct_service_factory! {
 					grandpa::NetworkBridge::new(service.network()),
 				)?;
 
-				executor.spawn(voter);
+				executor.spawn(voter.select(service.on_exit()).then(|_| Ok(())));
```

Comment:
Can we not take the `on_exit` as parameter for `run_grandpa`? I think that makes it more easy to use and people who write their own node does not forget to select on exit.

Reply:
I've updated the code according to your suggestions, but FWIW I think I prefer the original version  because it's easier to check, we just make sure that all call-sites of `spawn` are exit-guarded. I think longer-term we should go with @rphmeier's suggestion though.